src/main/java/org/jboss/arquillian/testcontainers/TestContainersObserver.java

```diff
@@ -21,29 +23,43 @@
 public class TestContainersObserver {
     @Inject
     @ClassScoped
-    protected InstanceProducer<GenericContainer<?>> containerWrapper;
+    protected InstanceProducer<TestContainerInstances> containersWrapper;
 
     public void createContainer(@Observes(precedence = 500) BeforeClass beforeClass) {
         TestClass javaClass = beforeClass.getTestClass();
+        TestContainers tcAnnos = javaClass.getAnnotation(TestContainers.class);
         TestContainer tcAnno = javaClass.getAnnotation(TestContainer.class);
-        if (tcAnno != null) {
-            checkForDocker(tcAnno.failIfNoDocker());
+        if (tcAnno != null || tcAnnos != null) {
+            TestContainer[] testContainers = (tcAnnos != null) ? tcAnnos.value() : new TestContainer[] { tcAnno };
+            List<GenericContainer<?>> containers = new ArrayList<>();
+            for (TestContainer testContainer : testContainers) {
+                checkForDocker(testContainer.failIfNoDocker());
```

Comment:
We don't need to do this every time as the first check regardless of the setting should fail if docker is not available.

Reply:
The only ... weirdness I can see is that the failIfNoDocker value is specified on each @Testcontainer instance, so it's possible that value could vary (assuming we go with repeatable annotations. I'll comment on that now)

Reply:
It won't really matter though. Say the first one is set to `false` and docker is not available, then an `AssertionError` is thrown and subsequent containers are ignored. If the first one is `true`, then an attempt to throw an assumption violation is thrown, and again subsequent containers are ignored.

Reply:
I had a look, and reworked it to an array in the annotation. 

To my suprise (after all these years doing Java):

@TestContainer(value = SimpleTestContainer.class, failIfNoDocker = false)

and 

@TestContainer({ HelloWildflyTest.MyPostgreSQLContainer.class, WildflyContainer.class })

are both valid Java code, when the value in the Annotation is defined as an array.
Which is nice for the simple case when you have 1 container (no array definition required) simple, the multiple containers are ok, and the failIfNoDocker is only declared once and therefore has a clear semantic for multiple containers

Reply:
Indeed. It only works on annotations, but it's nice to have it.